build.cake

```diff
@@ -92,21 +93,27 @@ Task("NuGetPublish")
 Task("Default")
 	.IsDependentOn("Test");
 
-void GenerateDocs(bool verify)
+void GenerateDocs(string docsAssembly, string docsSourceUri, bool verify)
 {
-	int exitCode = StartProcess(@"cake\XmlDocMarkdown\tools\XmlDocMarkdown.exe",
-		$@"{docsAssembly} docs\ --source ""{docsSourceUri}"" --newline lf --clean" + (verify ? " --verify" : ""));
+	var exePath = File("cake/XmlDocMarkdown/tools/XmlDocMarkdown.exe").ToString();
+	var arguments = docsAssembly + " docs" + System.IO.Path.DirectorySeparatorChar + @" --source """ + docsSourceUri + @""" --newline lf --clean" + (verify ? " --verify" : "");
+	if (Context.Environment.Platform.IsUnix())
+	{
+		arguments = exePath + " " + arguments;
+		exePath = "mono";
+	}
+	int exitCode = StartProcess(exePath, arguments);
 	if (exitCode == 1 && verify)
 		throw new InvalidOperationException("Generated docs don't match; use -target=GenerateDocs to regenerate.");
 	else if (exitCode != 0)
-		throw new InvalidOperationException($"Docs generation failed with exit code {exitCode}.");
+		throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Docs generation failed with exit code {0}.", exitCode));
```

Comment:
I know, it's seriously ingrained, but I think `InvariantCulture` is overkill here, especially for an exception message.